13_sandboxes/test_case_generator.py

```diff
@@ -0,0 +1,274 @@
+
+
+model_image = (
+    modal.Image.from_registry("nvidia/cuda:12.4.0-devel-ubuntu22.04", add_python="3.12")
+    .pip_install(
+        "transformers==4.53.2",
+        "torch==2.7.1",
+        "accelerate==1.8.1",
+        "hf_transfer==0.1.9",
+    )
+    .env(
+        {
+            "HF_HUB_ENABLE_HF_TRANSFER": "1",
+            "HF_HOME": "/cache",
+        }
+    )
+    .entrypoint([])  # silence noisy logs
+)
+
+with model_image.imports() as imports:
+    from transformers import AutoModelForCausalLM, AutoTokenizer
+
+
+# GH Repo Configs
+GH_OWNER = "modal-labs"
+GH_REPO_NAME = "password-analyzer"
+GH_MODULE_NAME = "password_strength"
+GH_BRANCH = "main"
+
+
+@app.cls(
+    image=model_image,
+    volumes={
+        "/cache": model_volume,
+        "/data": files_volume,
+    },
+    gpu="L40S",
+)
+class Deepseek:
+    @modal.enter()
+    def load_model(self):
+        MODEL_NAME = "deepseek-ai/deepseek-coder-6.7b-instruct"
+        REVISION = "e5d64addd26a6a1db0f9b863abf6ee3141936807"
+        self.tokenizer = AutoTokenizer.from_pretrained(MODEL_NAME, revision=REVISION)
+        self.model = AutoModelForCausalLM.from_pretrained(MODEL_NAME, revision=REVISION)
+        self.model.to("cuda")
+
+    def load_inputs(self, file_name: str) -> tuple[str, str]:
+        import os
+
+        if not os.path.exists("/data/inputs"):
+            raise Exception(
+                "Inputs directory does not exist. Make sure to run download_files_to_volume first."
+            )
+
+        with open(f"/data/inputs/{file_name}", "r") as f:
+            file_contents = f.read()
+
+        with open(f"/data/inputs/test_{file_name}", "r") as f:
+            test_file_contents = f.read()
+        return file_contents, test_file_contents
+
+    def write_outputs(self, output_file_name: str, output_contents: str) -> str:
+        import os
+
+        os.makedirs("/data/outputs", exist_ok=True)
+        with open(f"/data/outputs/{output_file_name}", "w") as f:
+            f.write(output_contents)
+        print(f"Output written to {output_file_name}")
+        return output_file_name
+
+    @modal.method()
+    def generate(self, file_name: str) -> str:
+        file_contents, test_file_contents = self.load_inputs(file_name)
+
+        system_prompt = get_system_prompt()
+        user_prompt = get_user_prompt(file_contents, test_file_contents)
+
+        messages = [
+            {"role": "system", "content": system_prompt},
+            {"role": "user", "content": user_prompt},
+        ]
+        print("Applying chat template...")
+        inputs = self.tokenizer.apply_chat_template(
+            messages, add_generation_prompt=True, return_tensors="pt"
+        ).to("cuda")
+        print("Generating...")
+        outputs = self.model.generate(
+            inputs,
+            max_new_tokens=1024,
+            do_sample=False,
+            num_return_sequences=1,
+        )
+        print("Decoding...")
```

Comment:
I'd skip this one -- it's a bit too inside-baseball to be worth sharing and it's fast.